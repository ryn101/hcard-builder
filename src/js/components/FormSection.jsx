import React from 'react';
import PropTypes from 'prop-types';

function FormSection(props) {
    const { title, children } = props;

    return (
        <section className="form__section">
            <h2 className="form__title">{title}</h2>
            { children }
        </section>
    );
}

FormSection.propTypes = {
    title: PropTypes.string.isRequired,
    children: PropTypes.oneOfType([
        PropTypes.element,
        PropTypes.arrayOf(PropTypes.element),
    ]).isRequired,
};

export default FormSection;
