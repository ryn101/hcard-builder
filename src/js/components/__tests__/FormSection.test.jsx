import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FormSection from '../FormSection';

configure({ adapter: new Adapter() });

it('renders the form section without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <FormSection title="Title">
            <span />
        </FormSection>,
        div,
    );

    ReactDOM.unmountComponentAtNode(div);
});

describe('once rendered', () => {
    let formSectionNode;
    const sectionTitle = 'The title it should have';
    const SectionChildrenType = 'span';
    const sectionChildren = [
        <SectionChildrenType key="1" />,
        <SectionChildrenType key="2" />,
    ];

    beforeEach(() => {
        formSectionNode = shallow(
            <FormSection title={sectionTitle}>
                {
                    sectionChildren
                }
            </FormSection>,
        );
    });

    it('renders with state correctly', () => {
        expect(formSectionNode).toMatchSnapshot();
    });

    it('renders the title correctly', () => {
        const titleNode = formSectionNode.find('.form__title');
        expect(titleNode.text()).toEqual(sectionTitle);
    });

    it('renders the children correctly', () => {
        const childNodes = formSectionNode.children(SectionChildrenType);
        expect(childNodes).toHaveLength(2);
    });
});
