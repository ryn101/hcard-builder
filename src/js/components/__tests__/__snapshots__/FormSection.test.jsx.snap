// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`once rendered renders with state correctly 1`] = `
<section
  className="form__section"
>
  <h2
    className="form__title"
  >
    The title it should have
  </h2>
  <span
    key="1"
  />
  <span
    key="2"
  />
</section>
`;
